src/demo/ui.jsx

```diff
@@ -596,11 +671,46 @@ class Pond extends React.Component {
       state.pondFish.length
     } that it classified as ${state.word.toUpperCase()}.`;
 
+    const showFishDetails = !!state.pondClickedFish;
+    let pondFishDetailsStyle;
+    let confidence;
+    if (showFishDetails) {
+      const fish = state.pondClickedFish;
+
+      const leftX = Math.min(
+        Math.max(state.pondClickedFish.x + 200, 20),
+        constants.canvasWidth - 210
+      );
+      const topY = Math.min(
+        Math.max(state.pondClickedFish.y, 20),
+        constants.canvasHeight - 50
+      );
+
+      pondFishDetailsStyle = {
+        ...styles.pondFishDetails,
+        left: leftX,
+        top: topY
+      };
+
+      if (!fish.confidence || !fish.confidence.confidencesByClassId) {
+        confidence = 'Not sure';
+      } else if (fish.confidence.confidencesByClassId[0] > 0.99) {
+        confidence = 'Very sure';
+      } else if (fish.confidence.confidencesByClassId[0] > 0.5) {
+        confidence = 'Fairly sure';
+      } else {
+        confidence = 'Not very sure';
+      }
```

Comment:
@uponthesun Very interested to know whether these heuristics are decent for now, and what they should be going forward...